activerecord/lib/active_record/attribute_methods.rb

```diff
@@ -44,11 +44,67 @@ def initialize_generated_modules # :nodoc:
         @generated_attribute_methods = const_set(:GeneratedAttributeMethods, GeneratedAttributeMethods.new)
         private_constant :GeneratedAttributeMethods
         @attribute_methods_generated = false
+        @alias_attributes_mass_generated = false
         include @generated_attribute_methods
 
         super
       end
 
+      def alias_attribute(new_name, old_name)
+        super
+
+        if @alias_attributes_mass_generated
+          ActiveSupport::CodeGenerator.batch(generated_attribute_methods, __FILE__, __LINE__) do |code_generator|
+            generate_alias_attribute_methods(code_generator, new_name, old_name)
+          end
+        end
+      end
+
+      def eagerly_generate_alias_attribute_methods(_new_name, _old_name) # :nodoc:
+        # alias attributes in Active Record are lazily generated
+      end
+
+      def generate_alias_attributes # :nodoc:
+        return if @alias_attributes_mass_generated
+
+        generated_attribute_methods.synchronize do
+          return if @alias_attributes_mass_generated
+          ActiveSupport::CodeGenerator.batch(generated_attribute_methods, __FILE__, __LINE__) do |code_generator|
+            attribute_aliases.each do |new_name, old_name|
+              generate_alias_attribute_methods(code_generator, new_name, old_name)
+            end
+          end
+
+          @alias_attributes_mass_generated = true
+        end
+      end
+
+      def alias_attribute_method_definition(code_generator, pattern, new_name, old_name)
+        method_name = pattern.method_name(new_name).to_s
+        target_name = pattern.method_name(old_name).to_s
+        parameters = pattern.parameters
+        old_name = old_name.to_s
+
+        method_defined_within = method_defined_within?(target_name, self)
+        manually_defined_on_class = method_defined_within && self.instance_method(target_name).owner == self
```

Comment:
Included modules count too. I think we might want something closer to `owner <= superclass` (`|| owner == generated_attribute_methods`? We should probably check that aliasing something _to_ an existing column name behaves the same before/after this change, though I have no idea what it would currently do.)

Reply:
>Included modules count too. 

You are right! And it made me think that the current version of the condition is overcomplicated. What we really want to check is that if anything apart from Rails itself has defined the `target_name` method. And since Rails defines its methods in `generated_attribute_methods` I'm going to try simplifying the check to 
`self.instance_method(target_name).owner != generated_attribute_methods` which would mean that the method has been manually defined by something in the application.

>We should probably check that aliasing something to an existing column name behaves the same before/after this change

I have to admit this would be an unusual usage of an alias attribute so perhaps Rails could treat it as a wrong usage of the feature but I'll see what happens now and what will happen after we make the change!

Reply:
Alright, I made some changes and now modules should be taken into consideration (covered by a test). Though I had to add an additional `dangerous_attribute_methods` check. I know that we tried to avoid adding logic that will explicitly mention a set of reserved terms but in this case I'd say we can justify the addition since it's only a temporary measure and `reserved_method_name` check will go away along with the deprecation. Our main logic that will stay doesn't need the `reserved_method_name` check.

Reply:
>aliasing something to an existing column name behaves the same 

I checked some basic usages and it works the same. Basically if we do `alias_attribute :existing_column_to_be_abandoned, :existing_column` - `existing_column_to_be_abandoned` becomes abandoned as any writes or accesses go to the other `existing_column`